### PR TITLE
Improve alignment in Visits > real-time by hiding parent of period selector

### LINKED ISSUE #19818: When viewing the real-time report, the Segment selector is not aligned
This is the case when the Matomo install has only one site. Then select "Visitors -> Real-time" in the reporting menu

image

The segment selector should be aligned with the card below and be moved a few pixels to the left.

### PR DESCRIPTION
**Pull request by peterhashair**
### Description:

Fixes:#19818

hide parent div of period selector in Visits->`real-time` to improve alignment. 

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: plugins/CoreHome/vue/src/PeriodSelector/PeriodSelector.vue [hunk 1]
```diff
@@ -250,7 +250,7 @@ export default defineComponent({
   },
   mounted() {
     Matomo.on('hidePeriodSelector', () => {
-      window.$(this.$refs.root as HTMLElement).hide();
+      window.$(this.$refs.root as HTMLElement).parent().hide();
```
Comment (diosmosis) [inline plugins/CoreHome/vue/src/PeriodSelector/PeriodSelector.vue]: maybe hide the parent only if it has [vue-entry]? this way if PeriodSelector is used directly in a Vue template, an element in the parent component wont be accidentally hidden

Comment (peterhashair) [reply plugins/CoreHome/vue/src/PeriodSelector/PeriodSelector.vue]: @diosmosis ah I found the issue case, the parent doesn't have [vue-entry], but should hide. maybe [vue-entry] doesn't work.

Comment (diosmosis) [reply plugins/CoreHome/vue/src/PeriodSelector/PeriodSelector.vue]: 👍

Comment (sgiehl) [reply plugins/CoreHome/vue/src/PeriodSelector/PeriodSelector.vue]: Doesn't the parent div always have the id `periodString` in case it should be hidden?

Comment (peterhashair) [reply plugins/CoreHome/vue/src/PeriodSelector/PeriodSelector.vue]: @sgiehl we probably can check if the id has `periodString` good point.

**Review by bx80 — APPROVED** [review]
Tests well, segment selector is properly aligned on the real time visits view and the period selector is re-enabled on other screens without issue. 

* [x] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [x] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [x] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [x] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [x] n/a [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [x] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [x] Tests updated [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [x] None [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [x] n/a [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [x] n/a [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [x] n/a Existing documentation updated if needed

**Pull request merged**